iocore/hostdb/HostDB.cc

```diff
@@ -1588,6 +1588,12 @@ HostDBContinuation::set_check_pending_dns()
   this->setThreadAffinity(this_ethread());
   HostDBContinuation *c = q.head;
   for (; c; c = static_cast<HostDBContinuation *>(c->link.next)) {
+    if (c == this) {
+      Warning("Skip the insertion of the same continuation to pending dns");
+      return false;
```

Comment:
One concern about this change is that if this continuation was the first request, returning false here will cause the caller to change the continuation handler.  There will be no continuation that will do the HostDBContinuation::dnsEvent.

Maybe the probeEvent handler deals with it, but I'm not sure.

Reply:
hmm. 🤔
The set_handler only changes handler for this continuation. Since the continuation is already in the pending dns list, there is no reason to do the dnsEvent() again for this continuation.
Could you help me to understand how this will affect the other requests?